cli/internal/lockfile/lockfile.go

```diff
@@ -0,0 +1,18 @@
+// Package lockfile provides the lockfile interface and implementations for the various package managers
+package lockfile
+
+import "io"
+
+// Lockfile Interface for general operations that work accross all lockfiles
+type Lockfile interface {
```

Comment:
As in provide a way to access the contents as a DAG? Are there any favorite graph libraries in Go world?

Reply:
Well, not as a DAG because `npm` allows cycles, but so that we're able to leverage basic graph algorithms for reduction. I think it'd be a worthwhile addition, but agreed not for this PR.

Reply:
Also: https://github.com/gonum/gonum/tree/master/graph